Apps/Src/FaultState.c

```diff
@@ -12,13 +12,26 @@
 #include "CANbus.h"
 #include "BSP_UART.h"
 #include "config.h"
-#ifndef SIMULATION
-#include "stm32f4xx.h"
-#else
+#include "BSP_PLL.h"
+#ifdef SIMULATION
 #include "Simulator.h"
 extern uint8_t stateCount;
 #endif
 
+void delay_u_copy(uint16_t micro)
+{
+  uint32_t delay = BSP_PLL_GetSystemClock() / 1000000;
+	for(uint32_t i = 0; i < micro; i++)
```

Comment:
Does this need to be volatile, or will the compiler not optimize out the loop?

Reply:
fixed